implementation/src/main/java/io/smallrye/graphql/bootstrap/Classes.java

```diff
@@ -38,59 +45,69 @@
  * @author Phillip Kruger (phillip.kruger@redhat.com)
  */
 public class Classes {
+    private static final Map<String, Class> loadedClasses = new HashMap<>();
 
     private Classes() {
     }
 
+    public static Class<?> loadClass(String className) {
+
+        if (isPrimative(className)) {
+            return getPrimativeClassType(className);
+        } else {
+            if (loadedClasses.containsKey(className)) {
+                return loadedClasses.get(className);
+            } else {
+                try {
+                    return AccessController.doPrivileged((PrivilegedExceptionAction<Class<?>>) () -> {
+                        ClassLoader loader = Thread.currentThread().getContextClassLoader();
+                        if (loader != null) {
+                            try {
+                                Class<?> c = Class.forName(className, false, loader);
+                                loadedClasses.put(className, c);
+                                return c;
+                            } catch (ClassNotFoundException cnfe) {
+                                // Let's try this class classloader.
+                            }
+                        }
+                        Class<?> c = Class.forName(className, false, Classes.class.getClassLoader());
+                        loadedClasses.put(className, c);
+                        return c;
+                    });
+                } catch (PrivilegedActionException pae) {
+                    throw new ClassloaderException("Can not load class [" + className + "]", pae);
+                }
```

Comment:
maybe this block could be extracted into a method, while the `loadedClasses.put(className, c);` happens here.

Reply:
Yes cool. loadedClasses.put can actually also happen in the method